packages/actor-init-sparql/test/system/Engine-test.ts

```diff
@@ -86,6 +89,68 @@ describe('System test: ActorInitSparql', () => {
         expect((await arrayifyStream((<IQueryResultBindings> await engine.query(query, context)).bindingsStream)))
           .toEqual([]);
       });
+
+      describe('custom function', () => {
+        const funcAllow = 'allowAll';
+        const baseFunctionName = (funcName: string) =>
```

Comment:
Can we also have another test case with a slightly more complex function?
For example, a function that converts all strings to uppercase?

Reply:
You'll probably need a BIND for this instead of FILTER